src/cargo/sources/registry/http_remote.rs

```diff
@@ -402,19 +518,33 @@ impl<'cfg> RegistryData for HttpRegistry<'cfg> {
         // reduces the number of connections done to a more manageable state.
         handle.pipewait(true)?;
 
-        // Make sure we don't send data back if it's the same as we have in the index.
         let mut headers = List::new();
+        headers.append("cargo-protocol: version=1")?;
```

Comment:
I haven't seen any discussion about what this `cargo-protocol` header is. Can you say more about it?

Reply:
It was a suggestion in the RFC process. It would enable the server to potentially do something different if we ever change how the protocol works. It currently doesn't do anything.

Reply:
Can you add a comment to that effect to explain what it is for?